src/server/status.rs

```diff
@@ -224,6 +266,127 @@ impl Status {
     }
 }
 
+impl RemoteStatus {
+    pub fn new(name: &str) -> Self {
+        Self {
+            name: name.into(),
+            host: None,
+            port: None,
+            user: None,
+            database: None,
+            version: None,
+            major_version: None,
+            status: RemoteStatusService::Unknown("uninitialized".into())
+        }
+    }
+
+    pub async fn probe(mut self, path: PathBuf) -> Self {
+        let deadline = Instant::now() + Duration::from_secs(1);
+        let builder = match async {
+            Ok::<Builder, anyhow::Error>(future::timeout(
+                deadline - Instant::now(),
+                Builder::read_credentials(&path)
```

Comment:
Generally we don't timeout disk functions. If your disk is super-slow you will not be able to use CLI or EdgeDB in any sensible way, so there is no need to put timeout here. And even less sense to subtract this time from the network time deadline because this will show network errors for perfectly working and responsive remote services.